src/components/TodoList.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable react/react-in-jsx-scope */
+import TodoNote from "./TodoNote";
+import "../index.css";
+import React, {useState, useEffect, useRef} from "react";
+import {nanoid} from "nanoid";
+import AddNote from "./AddNote";
+
+const TodoList = () => {
+  const getNotesFunction = () => {
+    const savedNotes = localStorage.getItem("react-notes-data");
+    return savedNotes ? JSON.parse(savedNotes) : [];
+  };
+
+  const [notes, setNotes] = useState(getNotesFunction);
+    const draggedOverNote = useRef();
+
+  const dragEnter = (noteToReplace) => {
+    draggedOverNote.note = noteToReplace;
+  };
+
+  const drop = (draggedNote) => {
+    let reorderedNotes = [...notes];
```

Comment:
Can you make this pass through the array only once?

Reply:
Right now the solution was to modify by index but this is only efficient if you are editing it in place.

Reply:
What do you mean by this?

Reply:
Basically right now I copy the existing array of notes into a new variable which takes O(N) time. Then I find the index of an element which would take O(N) per call which is called twice. This means you're passing over the array three times. 

Ideally I would like to only pass over the array once and get the updated list.

Reply:
I tried to use one loop for this and worked on it for some time today, but this seems to be breaking my code and I'm unable to swap the notes.

Reply:
Updated switched statement.